packages/go_router/test/src/go_router_material_test.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+import 'package:go_router/src/go_router_material.dart';
+
+void main() {
+  group('isMaterialApp', () {
+    testWidgets('returns [true] when MaterialApp is present', (tester) async {
+      final key = GlobalKey<_DummyStatefulWidgetState>();
+      await tester.pumpWidget(
+        MaterialApp(
+          home: DummyStatefulWidget(key: key),
+        ),
+      );
+      final isMaterial = isMaterialApp(key.currentContext! as Element);
+      expect(isMaterial, true);
+    });
+
+    testWidgets('returns [false] when CupertinoApp is present', (tester) async {
+      final key = GlobalKey<_DummyStatefulWidgetState>();
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: DummyStatefulWidget(key: key),
+        ),
+      );
+      final isMaterial = isMaterialApp(key.currentContext! as Element);
+      expect(isMaterial, false);
+    });
+  });
+
+  test('pageBuilderForMaterialApp creates a [MaterialPage] accordingly', () {
+    final key = UniqueKey();
+    const name = 'name';
+    const arguments = 'arguments';
+    const restorationId = 'restorationId';
+    const child = DummyStatefulWidget();
+    final page = pageBuilderForMaterialApp(
+      key: key,
+      name: name,
+      arguments: arguments,
+      restorationId: restorationId,
+      child: child,
+    );
+    expect(page.key, key);
```

Comment:
same here, maybe assert it is material app

Reply:
Same thing as above, the difference this time being that `pageBuilderForMaterialApp` returns a `MaterialPage<void>`.